docs/architecture/psa-crypto-repository.md

```diff
@@ -0,0 +1,17 @@
+PSA Cryptography repository
+===========================
+## Introduction
+
+The PSA Cryptography repository contains a reference implementation of the [PSA Cryptography API specification](https://armmbed.github.io/mbed-crypto/psa/#application-programming-interface).
+
+## Requirements
+
+* The PSA Cryptography repository exposes as public interface the cryptographic interface defined in the PSA Cryptography API specification and solely this interface.
```

Comment:
It is possible we might end up exposing some extra bits (PKCS maybe), but these will be small exceptional cases. We definitely won't expose the legacy interface.

Reply:
Which PKCS are you talking about?

“We definitely won't expose the legacy interface” vs “we might end up exposing some extra bits” — well, which one is it? That's a major design decision.

Reply:
I've changed that, including the unified driver interface and the on-going PSA crypto extensions.